### PR TITLE
gh-123254: Improve `tuple` C API docs with more info about errors

### PR DESCRIPTION
**Pull request by sobolevn**
* Issue: gh-123254

----
📚 Documentation preview 📚: https://cpython-previews--123255.org.readthedocs.build/

### CODE REVIEW

File: Doc/c-api/tuple.rst [hunk 1]
```diff
@@ -33,25 +33,28 @@ Tuple Objects
...
 
    Take a pointer to a tuple object, and return the size of that tuple.
+   or ``-1`` and set a :exc:`SystemError` exception if *op* is not a tuple.
 
 
 .. c:function:: Py_ssize_t PyTuple_GET_SIZE(PyObject *p)
 
-   Return the size of the tuple *p*, which must be non-``NULL`` and point to a tuple;
-   no error checking is performed.
+   Return the size of the tuple *op*, which must be non-``NULL``
```
Comment (erlend-aasland) [inline Doc/c-api/tuple.rst]:
```suggestion
   Return the size of the tuple *p*, which must be non-``NULL``
```

Comment (erlend-aasland) [reply Doc/c-api/tuple.rst]: Actually, it seems to me this particular change is not needed. Right?

File: Doc/c-api/tuple.rst [hunk 2]
```diff
@@ -141,6 +146,8 @@ type.
    Create a new struct sequence type from the data in *desc*, described below. Instances
    of the resulting type can be created with :c:func:`PyStructSequence_New`.
 
+   Returns ``NULL`` with an exception set on failure.
```
Comment (erlend-aasland) [inline Doc/c-api/tuple.rst]:
```suggestion
   Return ``NULL`` with an exception set on failure.
```

File: Doc/c-api/tuple.rst [hunk 3]
```diff
@@ -149,8 +156,8 @@ type.
 
 .. c:function:: int PyStructSequence_InitType2(PyTypeObject *type, PyStructSequence_Desc *desc)
 
-   The same as ``PyStructSequence_InitType``, but returns ``0`` on success and ``-1`` on
-   failure.
+   The same as ``PyStructSequence_InitType``, but returns ``0`` on success
```
Comment (erlend-aasland) [inline Doc/c-api/tuple.rst]:
```suggestion
   The same as :c:func:`PyStructSequence_InitType`, but return ``0`` on success
```

Comment (erlend-aasland) [reply Doc/c-api/tuple.rst]: Alternatively:

```suggestion
   As :c:func:`PyStructSequence_InitType`, but return ``0`` on success
```

... or:

```suggestion
   Like :c:func:`PyStructSequence_InitType`, but return ``0`` on success
```

File: Doc/c-api/tuple.rst [hunk 4]
```diff
@@ -207,6 +214,8 @@ type.
    Creates an instance of *type*, which must have been created with
    :c:func:`PyStructSequence_NewType`.
 
+   Returns ``NULL`` with an exception set on failure.
```
Comment (erlend-aasland) [inline Doc/c-api/tuple.rst]:
```suggestion
   Return ``NULL`` with an exception set on failure.
```

File: Doc/c-api/tuple.rst [hunk 5]
```diff
@@ -33,25 +33,28 @@ Tuple Objects
...
    Take a pointer to a tuple object, and return the size of that tuple.
+   or ``-1`` and set a :exc:`SystemError` exception if *op* is not a tuple.
 
 
 .. c:function:: Py_ssize_t PyTuple_GET_SIZE(PyObject *p)
 
-   Return the size of the tuple *p*, which must be non-``NULL`` and point to a tuple;
-   no error checking is performed.
+   Return the size of the tuple *op*, which must be non-``NULL``
+   and point to a tuple; no error checking is performed.
```
Comment (skirpichev) [inline Doc/c-api/tuple.rst]:
```suggestion
   Like :c:func:`PyTuple_Size`, but without error checking.
```

File: Doc/c-api/tuple.rst (hunk 3 above)
Comment (sobolevn) [reply Doc/c-api/tuple.rst]: I think in this context `returns` is correct: "the same as funcion, but (function) returns"

File: Doc/c-api/tuple.rst [hunk 6]
```diff
@@ -33,25 +33,27 @@ Tuple Objects
...
+   Return a new tuple object of size *n*,
+   or ``NULL`` with an exception set on failure. The tuple values
    are initialized to the subsequent *n* C arguments pointing to Python objects.
    ``PyTuple_Pack(2, a, b)`` is equivalent to ``Py_BuildValue("(OO)", a, b)``.
 
 
 .. c:function:: Py_ssize_t PyTuple_Size(PyObject *p)
 
    Take a pointer to a tuple object, and return the size of that tuple.
+   or ``-1`` and set a :exc:`SystemError` exception if *p* is not a tuple.
```
Comment (erlend-aasland) [inline Doc/c-api/tuple.rst]:
```suggestion
   On error, return ``-1`` and with an exception set.
```

File: Doc/c-api/tuple.rst [hunk 7]
```diff
@@ -33,25 +33,27 @@ Tuple Objects
...
 
 
 .. c:function:: Py_ssize_t PyTuple_GET_SIZE(PyObject *p)
 
-   Return the size of the tuple *p*, which must be non-``NULL`` and point to a tuple;
-   no error checking is performed.
+   Like :c:func:`PyTuple_Size`, but does no checking of its arguments.
```
Comment (erlend-aasland) [inline Doc/c-api/tuple.rst]:
```suggestion
   Like :c:func:`PyTuple_Size`, but without error checking.
```

**Review by erlend-aasland — APPROVED** [review]
Thanks!

**Pull request merged**